From 62a6c8168c124e0c4bde50ddaedcb1d19f64d85d Mon Sep 17 00:00:00 2001
From: Federico Mena Quintero <federico@gnome.org>
Date: Wed, 18 Jan 2023 13:58:24 -0600
Subject: [PATCH] (#104): Use meson 0.63 for GNOME 44

We can't use fs.copyfile() yet, since that is introduced until
meson 0.64.  Downgrade the idiom to configure_file().
---
 ci/container_builds.yml | 4 ++--
 devel-docs/meson.build  | 5 ++++-
 meson.build             | 2 +-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/ci/container_builds.yml b/ci/container_builds.yml
index dea74b3d..07d8899d 100644
--- a/ci/container_builds.yml
+++ b/ci/container_builds.yml
@@ -13,7 +13,7 @@ include:
 variables:
   # When branching change the suffix to avoid conflicts with images
   # from the main branch
-  BASE_TAG: "2022-12-13.3-main"
+  BASE_TAG: "2023-01-18.0-main"
   RUST_STABLE: "1.64.0"
 
 .container.opensuse@x86_64:
@@ -96,7 +96,7 @@ opensuse-container@x86_64:
       python3-pip
       systemd-devel
     FDO_DISTRIBUTION_EXEC: >-
-      pip3 install meson==0.64
+      pip3 install meson==0.63
 
 fedora-container@x86_64:
   extends:
diff --git a/devel-docs/meson.build b/devel-docs/meson.build
index 6ef4570d..d1b00219 100644
--- a/devel-docs/meson.build
+++ b/devel-docs/meson.build
@@ -75,7 +75,10 @@ docs_sources = [
 # and just do the build from there.
 copied_docs_sources = []
 foreach i: docs_sources
-  copied_docs_sources += fs.copyfile(i)
+  copied_docs_sources += configure_file(copy: true, input: i, output: i)
+
+  # When we bump meson to 0.64 or later, replace the line above with this:
+  # copied_docs_sources += fs.copyfile(i)
 endforeach
 
 custom_target(
diff --git a/meson.build b/meson.build
index 3200fc67..f659daf4 100644
--- a/meson.build
+++ b/meson.build
@@ -6,7 +6,7 @@ project('at-spi2-core', 'c',
           'warning_level=1',
           'c_std=c99',
         ],
-        meson_version: '>= 0.64.0')
+        meson_version: '>= 0.63.0')
 
 add_project_arguments([ '-D_POSIX_C_SOURCE=200809L', '-D_DEFAULT_SOURCE' ], language: 'c')
 
-- 
GitLab

